src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -50,10 +50,22 @@
 #define MPIDI_OFI_CONTEXT_MASK       (((1ULL << MPIDI_OFI_CONTEXT_BITS) - 1) << (MPIDI_OFI_SOURCE_BITS + MPIDI_OFI_TAG_BITS))
 #define MPIDI_OFI_SOURCE_MASK        (((1ULL << MPIDI_OFI_SOURCE_BITS) - 1) << MPIDI_OFI_TAG_BITS)
 #define MPIDI_OFI_TAG_MASK           ((1ULL << MPIDI_OFI_TAG_BITS) - 1)
+/* This value comes from the fact that we use a uint32_t in
+ * MPIDI_OFI_send_handler to define the dest and that is the size we expect
+ * from the OFI provider for its immediate data field. */
+#define MPIDI_OFI_MAX_RANK_BITS      (MPIDI_OFI_SOURCE_BITS > 0 ? MPIDI_OFI_SOURCE_BITS : 32)
```

Comment:
The reason for this was because you might have 0 bits for the rank in the match bits but you have up to 32 bits in the immediate data. If we don't have this macro, we'd have to add this check everywhere else.